packages/project-disaster-trail/src/components/Game/RadialGauge.js

```diff
@@ -0,0 +1,78 @@
+/** @jsx jsx */
+import { Component } from "react";
+import PropTypes from "prop-types";
+import { jsx, css } from "@emotion/core";
+import { CircularProgressbar, buildStyles } from "react-circular-progressbar";
+
+const gaugeDefaultStyle = css`
+  transition: transform 1s;
+  position: absolute;
+  top: 10px;
+  left: 10px;
+  height: 60px;
+  width: 60px;
+
+  &.gauge-animate-style {
+    transform: scale(1.5);
+  }
+`;
+
+class RadialGauge extends Component {
+  state = {
+    percent: 0
+  };
+
+  componentDidUpdate(prevProps) {
+    const { animateGauge } = this.props;
+    if (prevProps.animateGauge !== animateGauge) {
+      if (animateGauge) {
+        this.setPercent(100);
+      } else {
+        this.setPercent(0);
+      }
+    }
+  }
+
+  setPercent = percent => {
+    this.setState(() => ({
```

Comment:
I think this could be shortened to just `this.setState({percent})`

Reply:
I was never quite clear on how much [asynchronicity](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) would be an issue for this which is why I normally use the functional approach. My best guess would be it only matters if you're relying on the previous state / props which I'm not here (or the other places I use setState in this PR). Anyway, do you have any knowledge on that?

Reply:
All that `async` in terms of `setState` means is that the state is not set immediately. It's set at the end of the current frame. So if you:
```
// in your constructor
this.state = {progress: 0};

// later, in a function
this.setState({progress: 1});
console.log(this.state.progress); // returns 0, not 1.
```

Sidenote: 
You can think of setState as more of a queue. Like, hey I want 'progress' to change - and react says Yup Cool, give me a second to see if anything else is changing. Then it gathers _all_ the setStates and sets them once at the very end of the frame, instead of updating the state a bunch of times during the frame. Remember, every change in state triggers a render, so this is really just to minimize the number of renders that occur to a single one.

At any rate both the setPercent method here as well as the function inside the setState are called immediately, so there's no need to wrap `setState` or pass it a function to derive the `progress` - you should just be able to use `this.setState({progress});` instead of `setPercent`.   

Let me know if you have any other questions about state.

Reply:
I've removed the other functional calls for `setState()` :)